src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -271,7 +273,7 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                         EventPipeSession session = null;
                         try
                         {
-                            session = diagnosticsClient.StartEventPipeSession(providerCollection, collectRundownEvents, (int)buffersize);
+                            session = diagnosticsClient.StartEventPipeSession(providerCollection, collectRundownEvents, (int)buffersize, rundownKeyword);
```

Comment:
If the runtime doesn't support custom rundown do you want to warn the user? If we don't warn users then I think we need to be more careful when telling people to use the gc-collect profile because it won't always be cheap depending on which runtime version is being used.

I think either option could be OK - one is more coding work now, the other is more care when recommending it to users. I'd lean towards a warning myself.

cc @Maoni0

Reply:
I think we should tell the users in the tool's output if the runtime version isn't new enough to support this so they'll know this captures more events than what gc-collect intends and the file size will be larger (but still go ahead with collecting). 

this should definitely also be mentioned in the docs.

Reply:
I was thinking about the policy around this.

Previously, I would like to push down the retry logic inside the `DiagnosticsClient`, and so I chose to err on the safe side to collect more events.

The reason I wanted to push down the logic to this layer is because `DiagnosticsClient` is the highest level where `dotnet-trace` and `dotnet-monitor` share code path. 

In my opinion, the ideal behavior for gc-collect is to drop the `GCSettingsEvent` in case the runtime does not support runtime keywords instead of incurring an expensive rundown.

Pushing the logic higher level up give us the opportunity to do so, I think I will do just that.

The cost of this, however, is to have duplicated code in dotnet-trace and dotnet-monitor. That's okay, hopefully it will be eventually be reconciled by https://github.com/dotnet/dotnet-monitor/issues/4606

Reply:
ahh you are absolutely right - we should just drop the GCSettings event if the runtime ver isn't new enough. I don't think I was thinking about the right thing last night :)